velox/dwio/parquet/reader/ParquetData.cpp

```diff
@@ -17,15 +17,62 @@
 #include "velox/dwio/parquet/reader/ParquetData.h"
 
 #include "velox/dwio/common/BufferedInput.h"
+#include "velox/dwio/parquet/common/ParquetBloomFilter.h"
 #include "velox/dwio/parquet/reader/ParquetStatsContext.h"
 
 namespace facebook::velox::parquet {
 
+using thrift::RowGroup;
+
+namespace {
+bool isFilterRangeCoversStatsRange(
+    common::Filter* filter,
+    dwio::common::ColumnStatistics* stats,
+    const TypePtr& type) {
+  switch (type->kind()) {
+    case TypeKind::BIGINT:
+    case TypeKind::INTEGER:
+    case TypeKind::SMALLINT:
+    case TypeKind::TINYINT: {
+      auto intStats =
+          dynamic_cast<dwio::common::IntegerColumnStatistics*>(stats);
+      if (!intStats)
+        return false;
+
+      int64_t min =
+          intStats->getMinimum().value_or(std::numeric_limits<int64_t>::min());
+      int64_t max =
+          intStats->getMaximum().value_or(std::numeric_limits<int64_t>::max());
+
+      switch (filter->kind()) {
+        case common::FilterKind::kBigintRange:
+          return static_cast<common::BigintRange*>(filter)->lower() <= min &&
+              max <= static_cast<common::BigintRange*>(filter)->upper();
```

Comment:
Is `testInt64Range` good for this check, and do we need to consider if null is allowed?

Reply:
Can't use `testInt64Range` since it tests for overlap of two ranges. Here, we need to check if the filter covers the full stats range.

Reply:
stats contains a hasNull flag, shall we check it against the bigint range?